extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -1264,6 +1268,39 @@ export const popupMessageListener = (request: Request, sessionStore: Store) => {
     return { tokenIdList: updatedTokenIdList };
   };
 
+  const getMigratableAccounts = async () => {
+    const keyIdList = await getKeyIdList();
+
+    const mnemonicPhrase = mnemonicPhraseSelector(sessionStore.getState());
+    const allAccounts = allAccountsSelector(sessionStore.getState());
+    const wallet = fromMnemonic(mnemonicPhrase);
```

Comment:
unfortunately, when users add new accounts, we don't properly differentiate between accounts that were generated from the mnemonic vs accounts that were imported via secret key. Because of this, we have to do this roundabout process of generating a bunch of accounts from the mnemonic and seeing if the accounts the users has stored belong to the mnemonic. Yuck.

I think this is a band-aid for right now and in the future we should properly tag mnemonic-derived accounts as they are added so we can easily identify them without having to do this

Reply:
ah yeah that's too bad, makes sense 👍